src/idl_gen_js_ts.cpp

```diff
@@ -1255,36 +1259,59 @@ class JsTsGenerator : public BaseGenerator {
         code += "};\n\n";
       }
 
+      // Generate a convenient CreateX function
+      std::string paramDoc =
+          GenTypeAnnotation(kParam, "flatbuffers.Builder", "builder");
+      for (auto it = struct_def.fields.vec.begin();
+           it != struct_def.fields.vec.end(); ++it) {
+        const auto &field = **it;
+        if (field.deprecated) continue;
+        paramDoc += GenTypeAnnotation(kParam, GetArgType(field), GetArgName(field));
+      }
+      paramDoc += GenTypeAnnotation(kReturns, "flatbuffers.Offset", "", false);
+
+      GenDocComment(code_ptr, paramDoc);
+
       if (lang_.language == IDLOptions::kTs) {
-          // Generate a convenient CreateX function
           code += "static create" + Verbose(struct_def);
           code += "(builder:flatbuffers.Builder";
-          
-          for (auto it = struct_def.fields.vec.begin();
-               it != struct_def.fields.vec.end(); ++it) {
-            const auto &field = **it;
-            if (field.deprecated)
-              continue;
-
-            code += ", " + GetArgName(field) + ":" + GetArgType(field);
-          }
+      } else {
+        code += object_name + ".create" + Verbose(struct_def);
+        code += " = function(builder";
+      }
+      for (auto it = struct_def.fields.vec.begin();
+           it != struct_def.fields.vec.end(); ++it) {
+        const auto &field = **it;
+        if (field.deprecated) continue;
 
-          code += "):flatbuffers.Offset {\n";
-          code += "  " + struct_def.name + ".start" + Verbose(struct_def) + "(builder);\n";
+        if (lang_.language == IDLOptions::kTs) {
+          code += ", " + GetArgName(field) + ":" + GetArgType(field);
+        } else {
+          code += ", " + GetArgName(field);
+        }
+      }
 
-          for (auto it = struct_def.fields.vec.begin();
-               it != struct_def.fields.vec.end(); ++it) {
-              const auto &field = **it;
-              if (field.deprecated)
-                continue;
+      if (lang_.language == IDLOptions::kTs) {
+        code += "):flatbuffers.Offset {\n";
+        code += "  " + struct_def.name + ".start" + Verbose(struct_def) + "(builder);\n";
+      } else {
+        code += ") {\n";
+        code += "  " + object_name + ".start" + Verbose(struct_def) + "(builder);\n";
+      }
 
-              code += "  " + struct_def.name + ".add" + MakeCamel(field.name) +"(";
-              code += "builder, " + GetArgName(field) + ");\n";
-          }
+      std::string methodPrefix =
+          lang_.language == IDLOptions::kTs ? struct_def.name : object_name;
+      for (auto it = struct_def.fields.vec.begin();
+           it != struct_def.fields.vec.end(); ++it) {
+        const auto &field = **it;
+        if (field.deprecated) continue;
```

Comment:
Please move "continue" to a new line.

Reply:
Updated.

Reply:
Still in the same line.

Reply:
Updated.  Sorry about all of these formatting issues, my editor found a formatting rule set and keeps applying the rules to this file.